src/mono/mono/mini/aot-compiler.c

```diff
@@ -14355,6 +14392,217 @@ add_interp_in_wrapper_for_sig (MonoAotCompile *acfg, MonoMethodSignature *sig)
 	add_method (acfg, wrapper);
 }
 
+//---------------------------------------------------------------------------------------
+//
+// is_direct_pinvoke_parsable checks whether the direct pinvoke should be parsed into
+// the corresponding module and entrypoint names. Empty lines and comments are skipped.
+//
+// Arguments:
+//  * direct_pinvoke - the string corresponding to the direct pinvoke to parse
+//
+// Return Value:
+//  gboolean pertaining to whether or not the direct pinvoke should be parsed.
+//  into the respective module_name and entrypoint_name.
+//
+
+static gboolean
+is_direct_pinvoke_parsable (char *direct_pinvoke)
+{
+	if (!direct_pinvoke)
+		return FALSE;
+
+	if (strlen (direct_pinvoke) == 0)
+		return FALSE;
+
+	if (direct_pinvoke[0] == '#')
+		return FALSE;
+
+	return TRUE;
+}
+
+typedef enum {
+	DIRECT_PINVOKE_SKIP,
+	DIRECT_PINVOKE_PARSED,
+	DIRECT_PINVOKE_ERR,
+} DirectPInvokeParsedResult;
+
+//---------------------------------------------------------------------------------------
+//
+// parsed_direct_pinvoke parses the direct pinvoke for the module_name and entrypoint_name
+// It presumes that is_direct_pinvoke_parsable (direct_pinvoke) returned true.
+//
+// Parsing:
+// The direct pinvoke is stripped of any leading or trailing whitespace.
+// The remaining is assumed to be of the form MODULE or MODULE!ENTRYPOINT.
+// The passed in pointers are then reassigned to the corresponding strings.
+//
+// Arguments:
+//  * acfg - the MonoAotCompiler instance
+//  * dpi (direct pinvoke) - the string corresponding to the direct pinvoke to parse
+//  ** module_name_ptr - the pointer to the module name string
+//  ** entrypoint_name_ptr - the pointer to the entrypoint name string
+//
+// Return Value:
+//  enum pertaining to whether or not the direct pinvoke entry should be skipped, was
+//  successfully split into the respective module_name and entrypoint_name, or if
+//  an error has occurred.
+//
+
+static DirectPInvokeParsedResult
+parsed_direct_pinvoke (MonoAotCompile *acfg, char *dpi, char **module_name_ptr, char **entrypoint_name_ptr)
+{
+	DirectPInvokeParsedResult res = DIRECT_PINVOKE_PARSED;
+	char *direct_pinvoke = g_strdup (dpi);
+	if (!direct_pinvoke) {
+		aot_printerrf (acfg, "Failed to strdup the direct pinvoke '%s'.\n", dpi);
+		res = DIRECT_PINVOKE_ERR;
+		goto early_exit;
+	}
+
+	if (!g_strstrip (direct_pinvoke)) {
```

Comment:
Shouldn't the string already been stripped even done before calling is_direct_pinvoke_parsable?

Reply:
I agree, the order might not matter between the trimming step and checking for comments/empty lines. NativeAOT had it in the order of checking  for comments, trimming, then checking for empty lines. Would it add value to have that helper function or is `(direct_pinvoke && g_strstrip (direct_pinvoke))` as suggested in https://github.com/dotnet/runtime/pull/79721#discussion_r1054176973 sufficient?

I was able to remove the enum by relocating the `g_strstrip`